node/p2p/blossomsub.go

```diff
@@ -704,28 +692,38 @@ func initDHT(
 		panic(err)
 	}
 
+	idService := h.(interface{ IDService() identify.IDService }).IDService()
 	reconnect := func() {
 		wg := &sync.WaitGroup{}
 		defer wg.Wait()
-		for _, peerinfo := range bootstrappers {
-			peerinfo := peerinfo
+		for _, p := range bootstrappers {
+			p := p
+			logger := logger.With(zap.String("peer_id", p.ID.String()))
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
-				if peerinfo.ID == h.ID() ||
-					h.Network().Connectedness(peerinfo.ID) == network.Connected ||
-					h.Network().Connectedness(peerinfo.ID) == network.Limited {
+				if p.ID == h.ID() ||
+					h.Network().Connectedness(p.ID) == network.Connected ||
+					h.Network().Connectedness(p.ID) == network.Limited {
+					logger.Debug("peer already connected")
 					return
 				}
 
-				if err := h.Connect(ctx, peerinfo); err != nil {
+				h.Peerstore().AddAddrs(p.ID, p.Addrs, peerstore.AddressTTL)
```

Comment:
We use `DialPeer` instead of `Connect` because it actually gives us the newly dialed connection, and we can run a health check on it via the `identify` service. This ensures that the connection is not 'connected but not really'.